src/prommis/uky/uky_flowsheet.py

```diff
@@ -172,72 +169,88 @@
 from idaes.models.unit_models.separator import (
     EnergySplittingType,
     Separator,
     SplittingType,
+    SeparatorInitializer,
 )
 from idaes.models.unit_models.solid_liquid import SLSeparator
 from idaes.models_extra.power_generation.properties.natural_gas_PR import (
     EosType,
     get_prop,
 )
+import idaes.logger as idaeslog
 
+from prommis.leaching.leach_train import LeachingTrain
 from prommis.leaching.leach_reactions import CoalRefuseLeachingReactions
 from prommis.leaching.leach_solids_properties import CoalRefuseParameters
 from prommis.leaching.leach_solution_properties import LeachSolutionParameters
 from prommis.precipitate.precipitate_liquid_properties import AqueousParameter
 from prommis.precipitate.precipitate_solids_properties import PrecipitateParameters
 from prommis.precipitate.precipitator import Precipitator
 from prommis.roasting.ree_oxalate_roaster import REEOxalateRoaster
-from prommis.solvent_extraction.ree_aq_distribution import REESolExAqParameters
 from prommis.solvent_extraction.ree_og_distribution import REESolExOgParameters
 from prommis.solvent_extraction.solvent_extraction import SolventExtraction
-
 from prommis.uky.costing.ree_plant_capcost import QGESSCosting, QGESSCostingData
 
+_log = idaeslog.getLogger(__name__)
+
 
 def main():
     """
     Run the flowsheet by calling the appropriate functions in series.
     """
     m = build()
+    set_partition_coefficients(m)
 
     set_operating_conditions(m)
 
     scaled_model = set_scaling(m)
-    assert_units_consistent(scaled_model)
-    assert degrees_of_freedom(scaled_model) == 0
 
-    print("Structural issues after setting operating conditions")
-    dt = DiagnosticsToolbox(model=scaled_model)
-    dt.report_structural_issues()
+    if degrees_of_freedom(scaled_model) != 0:
+        raise AssertionError(
+            "The degrees of freedom are not equal to 0."
+            "Check that the expected variables are fixed and unfixed."
+            "For more guidance, run assert_no_structural_warnings from the IDAES DiagnosticToolbox "
+        )
 
     initialize_system(scaled_model)
-    print("Numerical issues after initialization")
-    dt.report_numerical_issues()
 
-    results = solve(scaled_model)
-    print("Numerical issues after solving")
-    dt.report_numerical_issues()
+    solve_system(scaled_model)
+
+    # fixes the volumetric flow rate of the organic recycle streams and unfixes the flow of the make-up streams
+    # we want to be able to adjust the total recycle flow rate, not just the make-up portion of it
+    fix_organic_recycle(scaled_model)
+
+    scaled_results = solve_system(scaled_model)
 
-    display_results(scaled_model)
+    if not check_optimal_termination(scaled_results):
+        raise RuntimeError(
+            "Solver failed to terminate with an optimal solution. Please check the solver logs for more details"
+        )
+
+    scaling = TransformationFactory("core.scale_model")
```

Comment:
I suggest moving the actual scaling transformation out of `set_scaling` - for one, the method name only implies setting scaling factors, not getting a scaled model. Secondly, this would avoid having to create a second instance of the scaling transformation here (whilst it doesn't really matter, it makes a lot more sense to use the same scaling transformation object to get the results back).